gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -275,12 +275,27 @@ private WriteObjectResponse doResumableUpload() throws IOException {
 
       boolean objectFinalized = false;
       while (!objectFinalized) {
+
+        if (responseObserver.isComplete()) {
+          // reset streams
+          responseObserver = new InsertChunkResponseObserver(uploadId, writeOffset);
+          call =
+              storageStub
+                  .getChannel()
+                  .newCall(StorageGrpc.getWriteObjectMethod(), stub.getCallOptions());
+          writeObjectRequestStreamObserver =
+              ClientCalls.asyncClientStreamingCall(call, responseObserver);
+          requestStreamObserver =
+              watchdog.watch(
```

Comment:
With this, the watchdog will be watching the previous call and the current call?

Reply:
No previous stream will be removed from the open stream as soon as we have encountered either error in `responseObserver` or `onComplete`.

Reply:
By call you mean stream? If so, No. Previous stream must have reached it's terminal state. We are making sure of it by checking state of responseObserver.

Reply:
do we want to add an assert for the same here?